DSpace/dspace-angular
# Use keyboard to select values

## References
* Fixes #2500
* Fixes #2805 

## Description
These changes allow for keyboard navigation in submission dropdowns

## Instructions for Reviewers
This PR modifies the `DsDynamicScrollableDropdownComponent` to support keyboard navigation. This is done by calling the already existing `getVocabularyEntriesByValue` instead of just calling `getVocabularyEntries` like done before.
Some helper methods where needed to handle the user input as we don't want to allow all characters. A method for handling user input already existed (`selectOnKeyDown`) so this was pretty straight forward.

Also moved the code from the `ngOnInit` method away to a reusable `loadOptions` method.

**how to test this PR**
This can be tested against the demo instance (demo.dspace.org) using the "type" dropdown in submission.

However I would recommend to test this locally and add a lot of values to the dropdown starting with the same letter(s) so you can verify that the scrolling / pagination functionality also works when filtering results.


## Checklist
